src/Processors/QueryPlan/JoinStepLogical.cpp

```diff
@@ -781,4 +785,100 @@ std::optional<ActionsDAG> JoinStepLogical::getFilterActions(JoinTableSide side,
     return {};
 }
 
+void JoinStepLogical::serializeSettings(QueryPlanSerializationSettings & settings) const
```

Comment:
Are these settings local for each step?

Reply:
It's hard to tell if all of them are local or not.
But I also think it's not too expensive to serialize all of them. (I had an idea that we might serialize only those that are different from the default (which can be sent separately); we can optimize it later if needed.)
For now, I prefer to keep it simple.